File: src/main/java/ru/serdtsev/homemoney/moneyoper/model/Label.java
package ru.serdtsev.homemoney.moneyoper.model;

import lombok.*;
import ru.serdtsev.homemoney.balancesheet.BalanceSheet;

import javax.persistence.*;
import java.util.UUID;

import static java.util.Objects.nonNull;

@Entity
@Table(name = "labels")
@Getter
@Setter
@ToString
@EqualsAndHashCode
@NoArgsConstructor
public class Label {
  @Id
  public UUID id;

  @ManyToOne
  @JoinColumn(name = "bs_id")
  public BalanceSheet balanceSheet;

  public String name;

  @Column(name = "root_id")
  public UUID rootId;

  @Column(name = "is_category")
  private Boolean isCategory;

  @Column(name = "cat_type")
  @Enumerated(EnumType.STRING)
  public CategoryType categoryType;

  @Column(name = "is_arc")
  public Boolean isArc;

  public Label(UUID id, BalanceSheet balanceSheet, String name) {
    this(id, balanceSheet, name, null, false, null);
  }

  public Label(UUID id, BalanceSheet balanceSheet, String name, UUID rootId, Boolean isCategory, CategoryType categoryType) {
    this.id = id;
    this.balanceSheet = balanceSheet;
    this.name = name;
    this.rootId = rootId;
    this.isCategory = isCategory;
    this.categoryType = categoryType;
    this.isArc = false;
  }

  public Boolean getIsCategory() {
    return nonNull(isCategory) ? isCategory : false;
  }
}

File: src/main/java/ru/serdtsev/homemoney/moneyoper/model/Label.kt
package ru.serdtsev.homemoney.moneyoper.model

import ru.serdtsev.homemoney.balancesheet.BalanceSheet
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "labels")
class Label(
        @Id
        var id: UUID?,

        @ManyToOne
        @JoinColumn(name = "bs_id")
        var balanceSheet: BalanceSheet?,

        var name: String?,

        @Column(name = "root_id")
        var rootId: UUID? = null,

        @Column(name = "is_category")
        var category: Boolean? = null,

        @Column(name = "cat_type")
        @Enumerated(EnumType.STRING)
        var categoryType: CategoryType? = null,

        @Column(name = "is_arc")
        val arc: Boolean? = null
) {
    constructor(id: UUID?, balanceSheet: BalanceSheet?, name: String?) : this(id, balanceSheet, name, null,
            null,null, null)

    fun isCategory() = category ?: false

    fun setCategory(value: Boolean) {
        category = value
    }
}